### PR TITLE
Expose a user's ongoing competitions

### LINKED ISSUE #6970: View user's ongoing competitions from WCA API
https://github.com/thewca/worldcubeassociation.org/blob/6e253ed58528223c789897d24909b1fbafba69fd/WcaOnRails/app/controllers/api/v0/api_controller.rb#L124

Right now we can query upcoming competitions but not their ongoing ones.

This would be super useful for applications ([such as mine](https://competition-groups.surge.sh/)) which show a user's groups for an upcoming competitions and it'd be nice to show the competition happening the same day.

I can likely send in a fix. This seems like adding something to the competition model and adding a query param?

### PR DESCRIPTION
**Pull request by coder13**
This calculates if a competition is ongoing by looking at if it has 1. started and 2. probably not over.
It then exposes this as a query param when fetching `/users/{userid}`

This directly fixes #6970 but I'd like to also, in a later PR, expose a route `/api/v0/users/{userid}/competitions` to fetch all of a user's competitions.

Alternatively, I could modify the `/api/v0/competitions` to add a query param to search for competitions with registrations containing a certain userId.

### CODE REVIEW

File: WcaOnRails/app/controllers/api/v0/api_controller.rb [hunk 1]
```diff
@@ -128,6 +128,13 @@ def show_user(user)
                                            .map(&:competition)
                                            .select(&:upcoming?)
       end
+      if params[:ongoing_competitions]
+        json[:ongoing_competitions] = user.registrations
+                                          .accepted
+                                          .includes(competition: [:delegates, :organizers, :events])
+                                          .map(&:competition)
+                                          .select(&:ongoing?)
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/api/v0/api_controller.rb:136]: The entire notion of querying a user's competitions is identical to `upcoming_competitions` above.

```ruby
user.registrations
.accepted
.includes(competition: [:delegates, :organizers, :events])
.map(&:competition)
```

Please _at least_ move that part out of the two if-blocks and only apply the `select` filter based on which query param is present.
Ideally, consider moving that entire block into the `user.rb` model as a kind of `accepted_competitions` method. (although that's arguably a bit out of scope of this PR, so I leave it to you to decide)

**Review by gregorbg — CHANGES_REQUESTED** [review]
Thanks for the quick implementation!
One nitpick about code reuse.

Also, please do not hesitate to pick up #6261 if you're interested, Saransh doesn't seem to be available to finish it right now.

And for the love of god please don't introduce a user ID filtering query param to the `/competitions` endpoint 😂

**Comment by coder13** [comment]
> And for the love of god please don't introduce a user ID filtering query param to the /competitions endpoint joy

haha, ok. Is this a performance thing?

Is it because you query for all competitions -> for all registrations where registrations contains user Id?

Just spitballing but we could even do a `/api/v0/competitions/mine` to mirror how it is in terms of the pages.

**Comment by gregorbg** [comment]
> > And for the love of god please don't introduce a user ID filtering query param to the /competitions endpoint joy
> 
> haha, ok. Is this a performance thing?
> 
> Is it because you query for all competitions -> for all registrations where registrations contains user Id?
> 
> Just spitballing but we could even do a `/api/v0/competitions/mine` to mirror how it is in terms of the pages.

It is mostly for avoiding feature duplication. I generally don't want to have the exact same information be available in two different ways, as that tends to confuse people who use the API externally.
We also currently don't have a really fleshed-out design spec, so I'd rather not take each and every idea about our API and implement them straight away.

**Review by gregorbg — APPROVED** [review]
Thanks for making the adjustments!

**Comment by coder13** [comment]
Thanks for merging!

**Pull request merged**